subprojects/core/src/main/java/org/gradle/initialization/GradleApiSpecAggregator.java

```diff
@@ -56,10 +56,10 @@ private Spec specFrom(Class<? extends GradleApiSpecProvider> provider) {
     }
 
     private Spec mergeSpecsOf(List<Class<? extends GradleApiSpecProvider>> providers) {
-        final ImmutableSet.Builder<Class<?>> exportedClasses = ImmutableSet.builder();
-        final ImmutableSet.Builder<String> exportedPackages = ImmutableSet.builder();
-        final ImmutableSet.Builder<String> exportedResources = ImmutableSet.builder();
-        final ImmutableSet.Builder<String> exportedResourcePrefixes = ImmutableSet.builder();
+        final ImmutableSet.Builder<Class<?>> exportedClasses = ImmutableSet.builderWithExpectedSize(providers.size());
```

Comment:
The expectation here is that every spec has only one exported class (package, etc.)? That does not seem to be realistic. Looking at `DefaultGradleApiSpecProvider`, that seems wrong.

Reply:
Ah, right, I missed that it used `addAll()` and not `add()`. Reverting.

Reply:
Ping me when you reverted the change.